### PR TITLE
Feature map 60 create form for find locations

### PR DESCRIPTION
**Pull request by romawht22**
Added filtering by tags and name

### CODE REVIEW

**Comment by steadyGuy** [comment]
Could u fix conflicts before review?

**Review by 6mahdapihka9 — APPROVED** [review]
👍

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]
image
after opening and closing filters list, checkboxes don't show which ones are checked and which ones are not.
also they don't work after reopening

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]
image
make the whole area inside green box clickable, not only red one


make filters list scrollable, even with this amount of filters some of them are under the footer

File: backend/controllers/LocationsController.ts [hunk 1]
```diff
@@ -1,40 +1,47 @@
...
-      );
-
-      return res.json({ locations });
+      if(name){
+        locations = locations.filter((l)=>{
+          return l.name.toLocaleLowerCase().startsWith(name)
+        })
+      }
+      if(filters.length > 0){
+        locations = locations.filter((l) => [...l.filters].some(el => filters.includes(el)));
```
Comment (steadyGuy) [inline backend/controllers/LocationsController.ts:37]: maybe, it should be done inside Location.find request?

File: backend/controllers/LocationsController.ts [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
-      locations = locations.slice(
-        0,
-        locations.length < 50 ? locations.length : 50
-      );
-
-      return res.json({ locations });
+      if(name){
+        locations = locations.filter((l)=>{
+          return l.name.toLocaleLowerCase().startsWith(name)
+        })
```
Comment (steadyGuy) [inline backend/controllers/LocationsController.ts:34]: A search implementation is much harder than that.
[one of implementations](https://www.digitalocean.com/community/tutorials/how-to-perform-full-text-search-in-mongodb)
Maybe, it should be a separate task though.

**Review by steadyGuy — APPROVED** [review]

File: backend/controllers/LocationsController.ts (hunk 2 above)
Comment (6mahdapihka9) [reply backend/controllers/LocationsController.ts:34]: yeap, we should create a task to change the filtering strategy

Comment (Yojuks) [reply backend/controllers/LocationsController.ts:34]: at next sprint - you should resolve this conversation

**Review by Yojuks — APPROVED** [review]

**Review by 6mahdapihka9 — APPROVED** [review]

**Pull request merged**